src/bin/julialauncher.rs

```diff
@@ -300,6 +302,181 @@ fn get_override_channel(
+            "--lisp",
+            "--image-codegen",
+            "--rr-detach",
+            "--strip-metadata",
+            "--strip-ir",
+            "--permalloc-pkgimg",
+            "--heap-size-hint",
+            "--trim",
+        ];
+        // `args` represents [switches...] -- [programfile] [programargs...]
+        // We want to find the first non-switch argument or the first argument after `--`
+        let mut skip_next = false;
+        for (i, arg) in args.iter().skip(1).enumerate() {
+            if skip_next {
+                skip_next = false;
+            } else if arg == "--" {
+                program_file = args.get(i + 1).cloned();
+                break;
+            } else if arg.starts_with("--") {
+                if !no_arg_long_switches.contains(&arg.as_str()) && !arg.contains('=') {
+                    skip_next = true;
+                }
+            } else if arg.starts_with("-") {
+                let arg: Vec<char> = arg.chars().skip(1).collect();
+                if arg.iter().all(|&c| no_arg_short_switches.contains(&c)) {
+                    continue;
+                }
+                for (j, &c) in arg.iter().enumerate() {
+                    if no_arg_short_switches.contains(&c) {
+                        continue;
+                    } else if j < arg.len() - 1 {
+                        break;
+                    } else {
+                        // `j == arg.len() - 1`
+                        skip_next = true;
+                    }
+                }
+            } else {
+                program_file = Some(arg.clone());
+                break;
+            }
+        }
+        if let Some(program_file) = program_file {
+            let mut path = PathBuf::from(program_file);
+            path.pop();
+            while !path.join("Project.toml").exists() && !path.join("JuliaProject.toml").exists() {
+                if !path.pop() {
+                    return None;
+                }
+            }
+            return Some(path);
+        } else {
+            return None;
+        }
+    } else if project.starts_with('@') {
+        let depot = match std::env::var("JULIA_DEPOT_PATH") {
+            Ok(val) => match val.split(':').next() {
+                Some(p) => PathBuf::from(p),
+                None => dirs::home_dir().unwrap().join(".julia"),
+            },
+            _ => dirs::home_dir().unwrap().join(".julia"),
+        };
+        let path = depot.join("environments").join(&project[1..]);
+        if path.exists() {
+            return Some(path);
+        } else {
+            return None;
+        }
+    } else {
+        return Some(PathBuf::from(project));
+    }
+}
+
+fn julia_version_from_manifest(path: PathBuf) -> Option<Version> {
+    let manifest = if path.join("Manifest.toml").exists() {
+        path.join("Manifest.toml")
+    } else if path.join("JuliaManifest.toml").exists() {
+        path.join("JuliaManifest.toml")
+    } else {
+        return None;
+    };
```

Comment:
`JuliaManifest.toml` should take priority over `Manifest.toml` as it does in Julia

Reply:
Yep, but we don't need to look for the version specific manifest files.

Reply:
Done.